app/controlplane/pkg/data/projectversion.go

```diff
@@ -109,6 +109,15 @@ func createProjectVersionWithTx(ctx context.Context, tx *ent.Tx, projectID uuid.
 		Save(ctx)
 }
 
+func findProjectVersionWithTx(ctx context.Context, tx *ent.Tx, projectID uuid.UUID, version string) (*ent.ProjectVersion, error) {
```

Comment:
It doesnt need to be `ent.Tx`, just use `ent.Client` and this method will be valid for bot transactional and not transactional clients.